src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -18,19 +18,321 @@
+            pumpStatus.Set(PumpStatus::kRemoteTemperature);
+            break;
+        case RemoteSensorType::kNoSensor:
+            // The pump is controlled by a setpoint, as defined by
+            // the ControlMode attribute. (N.B. The setpoint is an internal variable which MAY be
+            // controlled between 0% and 100%, e.g., by means of the Level Control cluster)
+            // The ControlMode can be any of the following:
+            // ConstantSpeed, ConstantPressure, ProportionalPressure,
+            // ConstantFlow, ConstantTemperature or Automatic. The actual ControlMode
+            // which would be the EffectiveControlMode is dependant on the actual
+            // physical pump application running "on-top" of this cluster server.
+            Attributes::EffectiveControlMode::Set(endpoint, controlMode);
+            pumpStatus.Clear(PumpStatus::kRemoteFlow);
+            pumpStatus.Clear(PumpStatus::kRemotePressure);
+            pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+            switch (controlMode)
+            {
+            case PumpControlMode::kConstantFlow:
+                pumpStatus.Set(PumpStatus::kRemoteFlow);
+                break;
+            case PumpControlMode::kConstantPressure:
+                pumpStatus.Set(PumpStatus::kRemotePressure);
+                break;
+            case PumpControlMode::kConstantTemperature:
+                pumpStatus.Set(PumpStatus::kRemoteTemperature);
+                break;
+            default:
+                // Intentionally left out
+                break;
+            }
+            break;
+        }
+        // Set the overall effective operation mode to Normal
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kNormal);
+    }
+    break;
+
+        // The pump is controlled by the OperationMode attribute.
+        // Maximum, Minimum or Local
+
+    case PumpOperationMode::kMaximum: {
+        uint8_t maxLevel;
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kMaximum);
+        Attributes::EffectiveControlMode::Set(endpoint, PumpControlMode::kConstantSpeed);
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
+        LevelControl::Attributes::MaxLevel::Get(endpoint, &maxLevel);
+        LevelControl::Attributes::CurrentLevel::Set(endpoint, maxLevel);
+#endif
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+
+    case PumpOperationMode::kMinimum: {
+        uint8_t minLevel;
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kMinimum);
+        Attributes::EffectiveControlMode::Set(endpoint, PumpControlMode::kConstantSpeed);
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
+        LevelControl::Attributes::MinLevel::Get(endpoint, &minLevel);
+        LevelControl::Attributes::CurrentLevel::Set(endpoint, minLevel);
```

Comment:
If MinLevel is 0 this does not work, since the Minimum control mode states that it is the lowest possible level without being stopped.

A simple check if MinLevel is 0 then set CurrentLevel to 1 else set it to MinLevel should be sufficient?

Reply:
Yes, I missed that :-)